### PR TITLE
Increment <locations> detector names

### PR DESCRIPTION
**Pull request by LamarMoore**
**Description of work.**

Added a new attribute which allows location names to be incremented.

**To test:**

Ensure unit test is sensible and passes.

*There is no associated issue.*

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: Framework/Geometry/src/Instrument/InstrumentDefinitionParser.cpp [hunk 1]
```diff
@@ -3008,6 +3008,12 @@ InstrumentDefinitionParser::convertLocationsElement(
         Strings::strip(pElem->getAttribute("name-count-start")));
   }
 
+  int nameCountIncrement(1);
```
Comment (OwenArnold) [inline Framework/Geometry/src/Instrument/InstrumentDefinitionParser.cpp]: Am I right in thinking that you support negative increments too? if not maybe make this unsigned?

File: docs/source/concepts/InstrumentDefinitionFile.rst [hunk 2]
```diff
@@ -958,7 +958,8 @@ The above one line of XML is shorthand notation for
 If name is specified, e.g. as name="det" in the first example, then as seen the
 <location> elements are given the 'name' plus a counter, where by default this counter starts from zero. This counter can optionally be changed by using
 attribute name-count-start, e.g. setting name-count-start="1" in the above
-example would have named the 10 <location> elements det1, det2, ..., det10.
+example would have named the 10 <location> elements det1, det2, ..., det10. Additionally, using the name-count-increment attribute, e.g setting
```
Comment (OwenArnold) [inline docs/source/concepts/InstrumentDefinitionFile.rst]: As per your code above, this also defaults to 1 (increment) if unspecified, should probably be mentioned here and tested for too.

**Review by OwenArnold — CHANGES_REQUESTED** [review]
Should probably say something about this in release notes too.

File: docs/source/concepts/InstrumentDefinitionFile.rst (hunk 2 above)
Comment (martyngigg) [reply docs/source/concepts/InstrumentDefinitionFile.rst]: I think we should  update XML schema too

File: docs/source/release/v4.1.0/idfs.rst [hunk 3]
```diff
@@ -0,0 +1,13 @@
+==================================
```
Comment (martyngigg) [inline docs/source/release/v4.1.0/idfs.rst]: Can we put these in `framework.rst`? We generally put the IDF changes in their own section in there tends not to be too many for its own page.

**Review by martyngigg — CHANGES_REQUESTED** [review]

File: docs/source/release/v4.1.0/index.rst [hunk 4]
```diff
@@ -76,6 +76,8 @@ Changes
   - :doc:`Direct Geometry <direct_geometry>`
 
   - :doc:`Indirect Geometry <indirect_geometry>`
+  
+  - :doc: `Instrument Definition Files <idfs>`
```
Comment (martyngigg) [inline docs/source/release/v4.1.0/index.rst]: I don't think we need this now that we've got rid `idfs.rst`?

**Review by martyngigg — APPROVED** [review]

**Review by OwenArnold — APPROVED** [review]

**Pull request merged**